bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -551,16 +551,27 @@ private static boolean isConvertibleToArrayType(Object sourceValue, BArrayType t
         }
         ArrayValue source = (ArrayValue) sourceValue;
         Type targetTypeElementType = targetType.getElementType();
-        if (source.getType().getTag() == TypeTags.ARRAY_TAG) {
-            Type sourceElementType = ((BArrayType) source.getType()).getElementType();
+        Type sourceType = source.getType();
+        if (sourceType.getTag() == TypeTags.ARRAY_TAG) {
+            Type sourceElementType = ((BArrayType) sourceType).getElementType();
             if (isNumericType(sourceElementType) && isNumericType(targetTypeElementType)) {
                 return true;
             }
         }
+        int targetSize = targetType.getSize();
+        long sourceSize = source.getLength();
+        boolean returnVal = true;
+        if (!TypeChecker.hasFillerValue(targetType) && sourceSize < targetSize) {
+            addErrorMessage(0, errors, "array cannot be expanded to size '" + targetSize + "' for target " +
+                    "type '" + targetType + "' without filler values");
+            returnVal = false;
```

Comment:
Because this will pass
```bal
public function main() {
    [int, int] intTuple = [1, 2];
    int[3] valueArray = checkpanic intTuple.cloneWithType();
}
```
and as the issue here is that the targetType not having filler values, I was thinking it is better to emphasise that by saying something like `array cannot be expanded to size '" + targetSize + "' because the target type '" + targetType + "' does not have a filler value."`

Another thing that needed to be fixed in this issue is in some cases the obtained filler values are not correct, according to the spec https://ballerina.io/spec/lang/master/#FillMember.
For example
```bal
public function main() {
    [string, string] tuple = ["aaa", "bbb"];
    string[3] arr = checkpanic tuple.cloneWithType(); // here arr[2] will become () where it should be ""
}
```